rfcs/text/testing-refinement.md

```diff
@@ -0,0 +1,143 @@
+
+One advantage of integrating Applitools is having the option of rendering the DOM snapshots in many other operating system, browser, device and viewport configurations we like by using their Ultrafast Grid. This does not mean though that we are testing our components in these other configurations at the time of test execution. There are other vendors, including CrossBrowserTesting and BrowserStack, which would allow us to run our required tests directly in these other configurations of real machines. However, this would mean a change of automation framework from Cypress to Selenium. This seems like a drastic change at this point as all the files, including the support files like helper files, indexes, locators etc, would need rewriting. In addition, not all of these other products are free for open source repositories as Applitools is and don’t all provide a dashboard which is as user-friendly as that of Applitools.
+
+# Adoption strategy
+
+Developers will need to amend how they write the stories. Examples of various main configurations of the components should be included in one story in the `Design System` directory without any knobs where possible. Existing stories will need to be amended to this format. Other React libraries such as Material-UI (https://material-ui.com) use this method of demonstrating their components. The current standard of documentation in the `Docs` tab will need to be maintained to provide a thorough description of how the component is meant to be used. In addition, these stories will need to be amended where necessary to show events in the `Actions` tab. Developers will need to write one or more additional stories in the `Test` directory for testing purposes presenting as many examples as possible of the components for visual regression purposes. The Cypress tests will need to be compared with the unit tests and, where tests covered by the regression suite can be moved into the unit tests, this should be done. Whilst some duplication will remain where event tests are concerned, we feel it is necessary to have these tests in Cypress too in order to provide a second layer of testing and further confidence that the action triggered by a user is the correct one, so these will remain.
```

Comment:
> Developers will need to write one or more additional stories in the Test directory for testing purposes presenting as many examples as possible of the components for visual regression purposes. 

I think we said that we were going to snapshot the MDX/Docs page which shows all examples of each component? The only reason to add a `Test` story is to do a visual regression or additional cypress (unit) test that we do not want to be in our Documentation.

Reply:
The current MDX/Docs pages don't show all examples of each component though. Like `Button`, hence why the 'All Buttons' story was created for testing purposes. Not all components have been updated to the format of the others in the Design System directory yet.`Icon` for example has only one example until you change the knobs and we can't do one snapshot for every single icon. Will every icon be shown once it's been refactored in the `Design System` directory? If not, then another story, like the 'All Buttons' story, will need to be written to show all icons where one single snapshot can be take. That's what I'm saying here.